fri/src/verifier/mod.rs

```diff
@@ -303,56 +303,24 @@ where
             mem::swap(&mut positions, &mut folded_positions);
         }
 
-        // 2 ----- verify the remainder of the FRI proof ----------------------------------------------
+        // 2 ----- verify the remainder polynomial of the FRI proof -------------------------------
+
+        // read the remainder polynomial from the channel and make sure it agrees with the evaluations
+        // from the previous layer.
+        let remainder = channel.read_remainder(max_degree_plus_1 - 1)?;
+        let offset: B = self.options().domain_offset();
 
-        // read the remainder from the channel and make sure it matches with the columns
-        // of the previous layer
-        let remainder_commitment = self.layer_commitments.last().unwrap();
-        let remainder = channel.read_remainder::<N>(remainder_commitment)?;
         for (&position, evaluation) in positions.iter().zip(evaluations) {
-            if remainder[position] != evaluation {
+            let comp_eval = eval_horner::<E>(
+                &remainder,
+                offset * domain_generator.exp((position as u64).into()),
+            );
+            if comp_eval != evaluation {
                 return Err(VerifierError::InvalidRemainderFolding);
             }
         }
```

Comment:
A couple of small nits here:
- I'd probably rename `remainder` into `remainder_poly` to indicate that it is a polynomial in coefficient form.
- I'd use `exp_vartime()` instead of `exp()` here to make exponentiation potentially faster.

Reply:
Done!